File: frontend-compose/src/main/kotlin/com/protonctl/app/model/SteamCandidate.kt
package com.protonctl.app.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class SteamInstallType {
    @SerialName("flatpak") FLATPAK,
    @SerialName("debian") DEBIAN,
    @SerialName("native") NATIVE,
    @SerialName("custom") CUSTOM,
    @SerialName("unknown") UNKNOWN,
}

@Serializable
data class SteamCandidate(
    val type: SteamInstallType,
    val path: String,
    val valid: Boolean,
    val label: String,
)

File: frontend/lib/models/steam_candidate.dart
enum SteamInstallType { flatpak, debian, native, custom, unknown }

SteamInstallType steamInstallTypeFromJson(String value) {
  switch (value) {
    case 'flatpak':
      return SteamInstallType.flatpak;
    case 'debian':
      return SteamInstallType.debian;
    case 'native':
      return SteamInstallType.native;
    case 'custom':
      return SteamInstallType.custom;
    default:
      return SteamInstallType.unknown;
  }
}

String steamInstallTypeToJson(SteamInstallType type) => type.name;

class SteamCandidate {
  SteamCandidate({
    required this.type,
    required this.path,
    required this.valid,
    required this.label,
  });

  factory SteamCandidate.fromJson(Map<String, dynamic> json) => SteamCandidate(
        type: steamInstallTypeFromJson(json['type'] as String? ?? 'unknown'),
        path: json['path'] as String? ?? '',
        valid: json['valid'] as bool? ?? false,
        label: json['label'] as String? ?? '',
      );

  final SteamInstallType type;
  final String path;
  final bool valid;
  final String label;
}
